File: frontend/src/types/stocks.ts
import type { TablePayload } from './common'

export interface StockDetail {
  stock_code: string
  name: string
  ranking: Record<string, unknown>
  market: Record<string, unknown>
  event_count: number
  found: boolean
}

export type StockHistory = TablePayload<Record<string, unknown>> & { stock_code: string }
export type StockEvidence = TablePayload<Record<string, unknown>> & { stock_code: string; query: string; warning?: string | null }
export interface StockExplanation { stock_code: string; available: boolean; cached: unknown; generated: false; message: string }

File: server/api/schemas/stocks.py
from __future__ import annotations

from typing import Any
from pydantic import BaseModel, Field


class StockDetailData(BaseModel):
    stock_code: str
    name: str = ""
    ranking: dict[str, Any] = Field(default_factory=dict)
    market: dict[str, Any] = Field(default_factory=dict)
    event_count: int = 0
    found: bool = False
